File: app/src/main/java/ai/elimu/analytics/dao/WordAssessmentEventDao.java
package ai.elimu.analytics.dao;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ai.elimu.analytics.entity.WordAssessmentEvent;

@Dao
public interface WordAssessmentEventDao {

    @Insert
    void insert(WordAssessmentEvent wordAssessmentEvent);

    @Query("SELECT * FROM WordAssessmentEvent ORDER BY time ASC")
    List<WordAssessmentEvent> loadAllOrderedByTimeAsc();

    @Query("SELECT * FROM WordAssessmentEvent ORDER BY time ASC")
    Cursor loadAllOrderedByTimeAscAsCursor();
}

File: app/src/main/java/ai/elimu/analytics/dao/WordAssessmentEventDao.kt
package ai.elimu.analytics.dao

import ai.elimu.analytics.entity.WordAssessmentEvent
import android.database.Cursor
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface WordAssessmentEventDao {
    @Insert
    fun insert(wordAssessmentEvent: WordAssessmentEvent)

    @Query("SELECT * FROM WordAssessmentEvent ORDER BY timestamp ASC")
    fun loadAllOrderedByTimestampAsc(): List<WordAssessmentEvent>

    @Query("SELECT * FROM WordAssessmentEvent ORDER BY timestamp DESC")
    fun loadAllOrderedByTimestampDesc(): Cursor

    @Query("SELECT * FROM WordAssessmentEvent WHERE wordId = :wordId ORDER BY timestamp DESC")
    fun loadAllOrderedByTimestampDesc(wordId: Long?): Cursor

    @Query("SELECT COUNT(*) FROM WordAssessmentEvent")
    fun getCount(): Int
}
